components/brave_sync/BUILD.gn

```diff
@@ -4,6 +4,20 @@ import("//tools/grit/grit_rule.gni")
 import("//tools/grit/repack.gni")
 import("buildflags/buildflags.gni")
 
+declare_args() {
+  brave_sync_endpoint = "https://sync-staging.brave.com"
```

Comment:
Should the default value point to the production?
As a default sync lib DEPS point to the production, so after doing clean checkout and run `npm run build` sync will not work.

Reply:
I'm confused about what this url should be because this points to staging as well https://github.com/brave/sync/blob/75c1bb10c4a54275de4fb10afd9f204cc07284a8/client/polyfill/chrome.js#L14

Reply:
why are we hardcoding the url here and in the library?

Reply:
We had Slack discussion and agreed we need to improve versioning and deployment of sync lib 
https://github.com/brave/sync/issues/364.